skimage/morphology/_skeletonize_3d_cy.pyx.in

```diff
@@ -0,0 +1,434 @@
+    includes it as number 13. The latter is consistent with [IAC15]_.
+    """
+    neighborhood[0] = img[p-1, r-1, c-1]
+    neighborhood[1] = img[p-1, r,   c-1]
+    neighborhood[2] = img[p-1, r+1, c-1]
+
+    neighborhood[ 3] = img[p-1, r-1, c]
+    neighborhood[ 4] = img[p-1, r,   c]
+    neighborhood[ 5] = img[p-1, r+1, c]
+
+    neighborhood[ 6] = img[p-1, r-1, c+1]
+    neighborhood[ 7] = img[p-1, r,   c+1]
+    neighborhood[ 8] = img[p-1, r+1, c+1]
+
+    neighborhood[ 9] = img[p, r-1, c-1]
+    neighborhood[10] = img[p, r,   c-1]
+    neighborhood[11] = img[p, r+1, c-1]
+
+    neighborhood[12] = img[p, r-1, c]
+    neighborhood[13] = img[p, r,   c]
+    neighborhood[14] = img[p, r+1, c]
+
+    neighborhood[15] = img[p, r-1, c+1]
+    neighborhood[16] = img[p, r,   c+1]
+    neighborhood[17] = img[p, r+1, c+1]
+
+    neighborhood[18] = img[p+1, r-1, c-1]
+    neighborhood[19] = img[p+1, r,   c-1]
+    neighborhood[20] = img[p+1, r+1, c-1]
+
+    neighborhood[21] = img[p+1, r-1, c]
+    neighborhood[22] = img[p+1, r,   c]
+    neighborhood[23] = img[p+1, r+1, c]
+
+    neighborhood[24] = img[p+1, r-1, c+1]
+    neighborhood[25] = img[p+1, r,   c+1]
+    neighborhood[26] = img[p+1, r+1, c+1]
+
+
+###### look-up tables
+def fill_Euler_LUT():
+    """ Look-up table for preserving Euler characteristic.
+
+    This is column $\delta G_{26}$ of Table 2 of [Lee94]_.
+    """
+    cdef int arr[128]
+    arr[:] = [1, -1, -1, 1, -3, -1, -1, 1, -1, 1, 1, -1, 3, 1, 1, -1, -3, -1,
+                 3, 1, 1, -1, 3, 1, -1, 1, 1, -1, 3, 1, 1, -1, -3, 3, -1, 1, 1,
+                 3, -1, 1, -1, 1, 1, -1, 3, 1, 1, -1, 1, 3, 3, 1, 5, 3, 3, 1,
+                 -1, 1, 1, -1, 3, 1, 1, -1, -7, -1, -1, 1, -3, -1, -1, 1, -1,
+                 1, 1, -1, 3, 1, 1, -1, -3, -1, 3, 1, 1, -1, 3, 1, -1, 1, 1,
+                 -1, 3, 1, 1, -1, -3, 3, -1, 1, 1, 3, -1, 1, -1, 1, 1, -1, 3,
+                 1, 1, -1, 1, 3, 3, 1, 5, 3, 3, 1, -1, 1, 1, -1, 3, 1, 1, -1]
+    cdef ndarray LUT = np.zeros(256, dtype=np.intc)
+    LUT[1::2] = arr
+    return LUT
+cdef int[::1] LUT = fill_Euler_LUT()
+
+
+# Fill the look-up table for indexing octants for computing the Euler
+# characteristic. See is_Euler_invariant routine below.
+{{py:
+_neighb_idx = [[2, 1, 11, 10, 5, 4, 14],      # NEB
+               [0, 9, 3, 12, 1, 10, 4],       # NWB
+               [8, 7, 17, 16, 5, 4, 14],      # SEB
+               [6, 15, 7, 16, 3, 12, 4],      # SWB
+               [20, 23, 19, 22, 11, 14, 10],  # NEU
+               [18, 21, 9, 12, 19, 22, 10],   # NWU
+               [26, 23, 17, 14, 25, 22, 16],  # SEU
+               [24, 25, 15, 16, 21, 22, 12],  # SWU
+              ]
+}}
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef bint is_Euler_invariant(pixel_type neighbors[],
+                             int[::1] lut):
+    """Check if a point is Euler invariant.
+
+    Calculate Euler characteristc for each octant and sum up.
```

Comment:
I started thinking "ugh, a text editor plugin for spell checking the comments only would be nice" --- only to discover that my text editor (gedit) *does* have exactly this plugin. Thanks for nudging me to look for it! Done, here and elsewhere.